src/v/coproc/event_handler.cc

```diff
@@ -0,0 +1,138 @@
+        /// deployed (this could occur during bootstrapping)
+        auto found = _active_ids.find(id);
+        if (event.header.action == event_action::remove) {
+            if (found != _active_ids.end()) {
+                disables.emplace_back(id);
+            }
+        } else {
+            /// ... or in the case if deploys are performed using the same
+            /// key. This would normally be resolved if the events came in
+            /// the same record batch by the reconcile events function, but
+            /// not if they arrived in separate batches.
+            if (found == _active_ids.end()) {
+                enables.emplace_back(enable_copros_request::data{
+                  .id = id, .source_code = std::move(event.data)});
+            }
+        }
+    }
+    /// TODO: In the future, maybe it would be cleaner to have a add/remove
+    /// endpoint, instead of two seperate RPC endpoints
+    if (!enables.empty()) {
+        enable_copros_request req{.inputs = std::move(enables)};
+        auto resp = co_await _dispatcher.enable_coprocessors(std::move(req));
+        if (resp.has_error()) {
+            vlog(
+              coproclog.error,
+              "Failed to register coprocessors with the wasm engine: {}",
+              resp.error());
+            throw event_handler_exception();
+            co_return;
+        } else {
+            for (script_id id : resp.value()) {
+                vlog(
+                  coproclog.info,
+                  "Successfully registered script with id: {}",
+                  id);
+                _active_ids.insert(id);
+            }
+        }
+    }
+    if (!disables.empty()) {
+        disable_copros_request req{.ids = std::move(disables)};
+        auto resp = co_await _dispatcher.disable_coprocessors(std::move(req));
+        if (resp.has_error()) {
+            vlog(
+              coproclog.error,
+              "Failed to make disable request to the wasm engine: {}",
+              resp.error());
+            /// In this case the code will follow a path that re-enters this
+            /// method with the same inputs, and the call to enable_copros
+            /// above succeeded but the call to disable_coprocessors failed,
+            /// double registrations will be avoided because the ids have
+            /// entered the \ref active_ids cache and will not be queued for
+            /// re-registration
+            throw event_handler_exception();
+        } else {
+            for (script_id id : resp.value()) {
+                _active_ids.erase(id);
+            }
+        }
+    }
+}
+
+ss::future<> data_policy_event_handler::process(
+  absl::btree_map<script_id, parsed_event> wsas) {
+    for (auto& [id, event] : wsas) {
+        if (event.header.action == event_action::deploy) {
+            auto [_, res] = _scritps.insert_or_assign(
+              id, std::move(event.data));
+            vassert(
+              res,
```

Comment:
After working with the design doc, i guess this is where you would eventually put in a place to handle these types of events in a future PR. So I guess for this PR it would be enough to get rid of this assert, and let the latest script with same id exist in the cache.

Reply:
Yes, yes 
It is by fault. I need to delete this assert

Reply:
done